Library/Model/SPIPrinting.m

```diff
@@ -0,0 +1,65 @@
+//
+//  SPIPrinting.m
+//  SPIClient-iOS
+//
+//  Created by Metin Avci on 14/9/18.
+//  Copyright © 2018 Assembly Payments. All rights reserved.
+//
+
+#import "SPIPrinting.h"
+#import "SPIRequestIdHelper.h"
+
+@implementation SPIPrintingRequest
+
+- (instancetype)initWithKey:(NSString *)key
+                    payload:(NSString *)payload{
+    
+    self = [super init];
+    
+    if (self) {
+        _infoId = [SPIRequestIdHelper idForString:@"print"];
+        _key = key;
```

Comment:
Normally you'd copy these strings to be safe from people passing in a NSMutableString, if you want to be super-careful, not a huge deal.
```suggestion
        _key = [key copy];
```